### PR TITLE
Clean District Column of Demog 1

### PR DESCRIPTION
**Pull request by as2388**
Review #11 first

### CODE REVIEW

File: survey_auto_code/survey_auto_code.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    # Load data from JSON file
+    with open(demog_1_input_path, "r") as f:
+        data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # TODO: Extend to add another argument for demog2, load this survey, and merge with demog1.
+    # TODO: Then we don't have to maintain parallel pipelines for data which was arbitrarily separated to begin with.
+
+    # Clean survey with regexes
+    district_key = "District (Text) - wt_demog_1"
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: consider DISTRICT_KEY

File: survey_auto_code/survey_auto_code.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    # Set missing entries in the raw data to 'NA'
+    for td in data:
+        for key in demog_keys:
+            long_key = "{} (Text) - wt_demog_1".format(key)
+            if long_key not in td:
+                td.append_data({long_key: "NA"}, Metadata(user, Metadata.get_call_location(), time.time()))
+
+    # Write json output
+    if os.path.dirname(json_output_path) is not "" and not os.path.exists(os.path.dirname(json_output_path)):
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: I do think we're starting to see this too much.... Consider ensure_path_exists(...)

File: survey_auto_code/survey_auto_code.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                {"{}_clean".format(district_key): somali.DemographicCleaner.clean_somalia_district(td[district_key])},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+
+    # Set missing entries in the raw data to 'NA'
+    for td in data:
+        for key in demog_keys:
+            long_key = "{} (Text) - wt_demog_1".format(key)
+            if long_key not in td:
+                td.append_data({long_key: "NA"}, Metadata(user, Metadata.get_call_location(), time.time()))
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: NA as const please

File: survey_auto_code/survey_auto_code.py (hunk 2 above)
Comment (as2388) [reply survey_auto_code/survey_auto_code.py]: Requires https://github.com/AfricasVoices/CoreDataModules/pull/39

**Pull request merged**